File: src/main/java/co/utb/softeng/moviesapp/entities/Movie.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package co.utb.softeng.moviesapp.entities;

import java.io.Serializable;
import java.util.Date;
import java.util.List;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

/**
 *
 * @author William
 */
@Entity
@Table(name = "MOVIES")
public class Movie implements Serializable {
    
    @Id
    @GeneratedValue
    private Long id;
    private String name;
    @Temporal(value=TemporalType.DATE)
    private Date releaseDate;
    private Double grossIncome;
    private String director;
    private double rating;
    private String genre;
    @ManyToMany(mappedBy="movies")
    private List<Actor> actors;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }
    
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Date getReleaseDate() {
        return releaseDate;
    }

    public void setReleaseDate(Date releaseDate) {
        this.releaseDate = releaseDate;
    }

    public Double getGrossIncome() {
        return grossIncome;
    }

    public void setGrossIncome(Double grossIncome) {
        this.grossIncome = grossIncome;
    }

    public String getDirector() {
        return director;
    }

    public void setDirector(String director) {
        this.director = director;
    }

    public double getRating() {
        return rating;
    }

    public void setRating(double rating) {
        this.rating = rating;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }

    public List<Actor> getActors() {
        return actors;
    }

    public void setActors(List<Actor> actors) {
        this.actors = actors;
    }
    
    
    
    
    
}

File: src/main/webapp/resources/assets/js/movie.js
/* 
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

$.ajax({
    type: 'GET',
    url: apiUrl + 'movie/' + movieId
}).done(function (movie) {
    var html = '';


    html += "<div class='movie'>Name:" + movie.name + "</a><br/>";
    html += "Director: " + movie.director + "<br/>";
    html += "Genre: " + movie.genre + "</div>";
    html += "<br/> <h1>Actors:</h1> <div id='actorsDiv'>";
    _.each(movie.actors, function (actor) {

        html += "<div class='movie'>First name: " + actor.firstName + "<br/>";
        html += "Last name: " + actor.lastName + "</div>";
        html += "Gender: " + actor.gender + "</div>";
        html += "Birth date: " + actor.birthdate + "</div>";

    });
    html += "</div>";


    $('#movieDiv').html(html);
});

